crates/re_chunk_store/src/query.rs

```diff
@@ -0,0 +1,236 @@
+                temporal_chunk_ids_per_timeline
+                    .iter()
+                    .filter(|(cur_timeline, _)| *cur_timeline == timeline)
+                    .flat_map(|(_, temporal_chunk_ids_per_component)| {
+                        temporal_chunk_ids_per_component.keys().copied()
+                    })
+                    .collect()
+            });
+
+        match (static_components, temporal_components) {
+            (None, None) => None,
+            (None, comps @ Some(_)) | (comps @ Some(_), None) => comps,
+            (Some(static_comps), Some(temporal_comps)) => {
+                Some(static_comps.into_iter().chain(temporal_comps).collect())
+            }
+        }
+    }
+
+    /// Check whether a given entity has a specific [`ComponentName`] either on the specified
+    /// timeline, or in its static data.
+    #[inline]
+    pub fn entity_has_component(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+        component_name: &ComponentName,
+    ) -> bool {
+        re_tracing::profile_function!();
+        self.all_components(timeline, entity_path)
+            .map_or(false, |components| components.contains(component_name))
+    }
+
+    /// Find the earliest time at which something was logged for a given entity on the specified
+    /// timeline.
+    ///
+    /// Ignores static data.
+    #[inline]
+    pub fn entity_min_time(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> Option<TimeInt> {
+        let temporal_chunk_ids_per_timeline =
+            self.temporal_chunk_ids_per_entity.get(entity_path)?;
+        let temporal_chunk_ids_per_component = temporal_chunk_ids_per_timeline.get(timeline)?;
+
+        let mut time_min = TimeInt::MAX;
+        for temporal_chunk_ids_per_time in temporal_chunk_ids_per_component.values() {
+            let Some(time) = temporal_chunk_ids_per_time
+                .per_start_time
+                .first_key_value()
+                .map(|(time, _)| *time)
+            else {
+                continue;
+            };
+            time_min = TimeInt::min(time_min, time);
+        }
+
+        (time_min != TimeInt::MAX).then_some(time_min)
+    }
+
+    /// Returns the most-relevant chunk(s) for the given [`LatestAtQuery`].
+    ///
+    /// The [`ChunkStore`] always work at the [`Chunk`] level (as opposed to the row level): it is
+    /// oblivious to the data therein.
+    /// For that reason, and because [`Chunk`]s are allowed to temporally overlap, it is possible
+    /// that a query has more than one relevant chunk.
+    ///
+    /// The caller should filter the returned chunks further (see [`Chunk::latest_at`]) in order to
+    /// determine what exact row contains the final result.
+    ///
+    /// If the entity has static component data associated with it, it will unconditionally
+    /// override any temporal component data.
+    pub fn latest_at(
```

Comment:
I find it hard to keep track of the meaning where there multiple levels of things called `latest_at`.

This one, in particular, would be more clearly named `latest_at_relevant_chunks` or equivalent.